mygit.rb

```diff
@@ -0,0 +1,365 @@
+        Dir.chdir ".."
+        $LOG.debug("Read data #{data} from #{file}")
+        return data
+
+
+end
+
+
+def mygit_prepare(base)
+        
+        repo = getreponame()
+        if base.nil? and !(Base?(repo))
+                $LOG.debug("<MyGit> <mygit_prepare> : Base Branch is not specified")
+                puts "Base branch not specified!"
+                exit
+        end
+        Dir.chdir(repo)
+        if base.nil?
+        $LOG.debug("<MyGit> <mygit_prepare> : Reading base from file as it is not provided in function parameter")
+        base = readhidden(repo, "Base.txt")
+        end
+
+        writetofile("Base.txt",base,repo)
+        
+        $LOG.debug("<MyGit> <mygit_prepare> : Calling rebase")
+        system "git rebase #{base}"
+        $LOG.debug("<MyGit> <mygit_prepare> : Rebase ended")
+        directory = readhidden(repo,"Direx.txt")
+
+        if $force == 'True'
+                system "git push -f -u origin #{directory}"
+        else
+                system "git push -u origin #{directory}"
+        end
+        $LOG.debug("<MyGit> <mygit_prepare> : git pushed #{directory}")
+end
+
+def mygit_request(reviewers)
+        $LOG.debug("Request Called")
+        $LOG.debug("<MYGIT> <mygit_request>module : Check if git exists? : #{GitExist?()}")
+        reviewers.shift
+        repo_directory = File.open("repo.txt")
+        repo_directory = repo_directory.read
+        repo = getreponame()
+        
+        Dir.chdir(repo)
+        
+        base_branch = readhidden(repo,"Base.txt")       
+
+        working_branch = readhidden(repo,"Direx.txt")
+
+	Dir.chdir('..')
+        client = octoKitInit()
+
+        create_pull = client.create_pull_request(repo_directory,base_branch, working_branch , "From Girish's App", "Pull Request body")
+        
+        reqreview=client.request_pull_request_review(repo_directory, create_pull.number , reviewers: reviewers)
+        
+        reviewers.each do |reviewer|        
+                puts "<MyGit> : Review requested by #{reviewer}"
+        end
+
+
+        comparison_branches = client.compare(repo_directory, base_branch ,working_branch , options = {})
+        comparison_json = Hash.new
+        counter=0
+        comparison_branches.files.each do |file|
+                array_files=[]  
+
+                file.each do |f|
+                        array_files.push(f)
+        
+                end
+
+                comparison_json[counter] = array_files
+                        counter=counter+1
+        end
+        File.open("comparison_result.json","w") do |f|
+                f.write(comparison_json.to_json)
+        end
+
+        writetofile("pullreq.txt", create_pull.number, repo)
+        writetofile("reviewreq.txt", reqreview.id , repo)                        
+  
+                                
+end
+
+def mygit_status()
+        client = octoKitInit()
+        repo = getreponame()
+        repo_directory = File.open("repo.txt").read
```

Comment:
All file opens should also close the file. Else handle operations in a block so ruby closes the file at the end of the block.